pinc/Project.inc

```diff
@@ -979,6 +981,27 @@ class Project
             || that_user_is_proj_facilitator($username));
     }
 
+    public function user_can_delete_nonpage_images($username)
+    {
+        if (is_null($username)) {
+            return false;
+        }
+        // SAs can always delete nonpage images
+        if (that_user_is_a_sitemanager($username)) {
+            return true;
+        }
+
+        // PMs/PFs can only delete nonpage images in New
+        // or P1 Unavailable stages
+        if ($this->can_be_managed_by_user($username)) {
+            return $this->state == PROJ_NEW || $this->state == PROJ_P1_UNAVAILABLE;
+        } else {
+            return false;
+        }
+        // Nobody else can delete images
+        return false;
```

Comment:
*minor* I'm not sure this line is ever reachable since all code paths should've entered at least one of the above ifs or elses. I'd just remove the else on line 999